### PR TITLE
KOGITO-5003: DMN editor removing edges for duplicate Decision Nodes on canvas

### PR DESCRIPTION
**Pull request by danielzhe**
See: [KOGITO-5003](https://issues.redhat.com/browse/KOGITO-5003)

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/unmarshall/nodes/NodeConnector.java [hunk 1]
```diff
@@ -517,15 +531,42 @@ private boolean isCentre(final double magnetRelativeX,
...
+    }
+
+    Node getTargetNode(final JSIDMNEdge jsidmnEdge,
+                       final List<NodeEntry> entries) {
+
+        if (entries.size() == 1) {
+            return entries.get(0).getNode();
+        }
+
+        final JSIPoint jsiTarget = Js.uncheckedCast(jsidmnEdge.getWaypoint().get(1));
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/unmarshall/nodes/NodeConnector.java:558]: I know there is a jira asking an enhancement to add "bend points" for connections. Unfortunately I struggle to find the jira link.

I guess we use index `1` to access target waypoint, what will maybe get out of date with implementing such a feature. Maybe could be replaced by something like `.size() - 1`?

Just mentioning to consider. Not blocking merge.

**Review by jomarko — APPROVED** [review]
thank you @danielzhe , I added @jstastny-cz as final reviewer to be aligned with jira.

Just single comment from me, code looks good to me, thank you.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/unmarshall/nodes/NodeConnector.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/marshaller/unmarshall/nodes/NodeConnector.java:558]: @jomarko here's the JIRA https://issues.redhat.com/browse/KOGITO-5119 - I agree `.size() - 1` would be more flexible, but I personally would keep it as is for now, as we will need to add other changes to support bend-points anyhow :-)

**Review by karreiro — COMMENTED** [review]
Thank you, @danielzhe!

I believe we maybe need some extra tweaks in the `NodeConnector.java` to cover this scenario:
Kapture 2021-05-24 at 10 45 26

Thanks again for the PR!

**Comment by danielzhe** [comment]
Great found, @karreiro ! I gonna check this issue. Thank you!

**Comment by danielzhe** [comment]
Changes applied, @karreiro

**Review by karreiro — APPROVED** [review]
Thank you, @danielzhe! Great stuff 🚀

**Comment by karreiro** [comment]
@jstastny-cz Could you please take a look at this PR? (I could not spot any issue involving multiple DRDs, but maybe you catch something I missed) 😅

**Comment by jstastny-cz** [comment]
jenkins do fdb

**Comment by jstastny-cz** [comment]
@karreiro I honestly don't know what and how to build in this repository. Are the artifacts available somewhere for the PR?

**Comment by karreiro** [comment]
@jstastny-cz I will build the (vsix) artifact, and left on this PR as soon as possible :-) (we're facing some issues to generate it locally based on a local path, but I'm coordinating it with the Foundation team, and as soon as possible I will generate it, and leave it here).

**Comment by karreiro** [comment]
@jstastny-cz Here's [the vsix plugin](https://www.dropbox.com/s/vuzwff3tgt9qrra/preview.vsix?dl=0) :-)

Here's the steps I performed to build it locally ;-)

```zsh
# Build 'kogito-editors-java' (1/4)"
cd "/Volumes/Navi/Projects/forks/kogito-editors-java"
mvn clean install -T 6 -Denforcer.skip=true -Dcheckstyle.skip=true -DskipTests=true -Dgwt.compiler.skip=true

# Build 'kie-wb-common-dmn-webapp-kogito-runtime' (2/4)"
cd "/Volumes/Navi/Projects/forks/kogito-editors-java/kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime"
mvn clean install -Denforcer.skip=true -Dcheckstyle.skip=true -DskipTests=true

export EXTERNAL_RESOURCE_PATH__dmnEditor=/Volumes/Navi/Projects/forks/kogito-editors-java/kie-wb-common-dmn/kie-wb-common-dmn-webapp-kogito-runtime/target/kie-wb-common-dmn-webapp-kogito-runtime

# Build 'kogito-tooling' (3/4)"
cd "/Volumes/Navi/Projects/forks/kogito-tooling"
git fetch upstream && git checkout master && git reset upstream/master --hard

yarn bootstrap
yarn build:prod

# Build .vsix plugin (4/4)"
cd "/Volumes/Navi/Projects/forks/kogito-tooling/packages/vscode-extension-pack-kogito-kie-editors"
yarn run package:prod
open /Volumes/Navi/Projects/forks/kogito-tooling/packages/vscode-extension-pack-kogito-kie-editors/dist/
```

**Comment by caponetto** [comment]
Hi guys,
Sorry to jump in but know that the FDB workflow was recently updated in https://github.com/kiegroup/kogito-editors-java/pull/54.
So, you need to rebase this PR with the `main` branch in order to get the updated workflows.
Also, keep in mind that the FDB workflow uploads the `vsix` file after successfully building `kogito-tooling` (even if IT tests fail).

**Comment by karreiro** [comment]
Thank you for the heads up, @caponetto! :-)

**Comment by jstastny-cz** [comment]
@danielzhe could you please rebase? I would prefer to test with FDB produced vsix instead of a local build.

**Review by jstastny-cz — APPROVED** [review]
LGTM, thanks.

**Pull request merged**